pkg/gofr/config/godotenv.go

```diff
@@ -24,30 +29,30 @@ func NewEnvFile(configFolder string, logger logger) *EnvLoader {
 }
 
 func (e *EnvLoader) read(folder string) {
-	var defaultFile = folder + "/.env"
-
-	overrideFile := func() string {
-		gofrEnv := e.Get("APP_ENV")
-		if gofrEnv != "" {
-			return fmt.Sprintf("%s/.%s.env", folder, gofrEnv)
-		}
+	var (
+		defaultFile  = folder + defaultFileName
+		overrideFile = folder + defaultOverrideFileName
+		env          = e.Get("APP_ENV")
+	)
 
-		return fmt.Sprintf("%s/.local.env", folder)
-	}()
+	if env != "" {
+		overrideFile = fmt.Sprintf("%s/.%s.env", folder, env)
+	}
 
 	// If 'APP_ENV' is set to x, then GoFr will try to read '.x.env' file from configs directory, if failed it will read
 	// the default file '.env'.
 	// If 'APP_ENV' is not set , GoFr will first try to read '.local.env', if failed then it
 	// will read default env file.
-	if err := godotenv.Load(overrideFile); err != nil {
+	err := godotenv.Load(overrideFile)
+	if err != nil && env != "" {
 		e.logger.Warnf("Failed to load config from file: %v, Err: %v", overrideFile, err)
-	} else {
+	} else if err == nil {
```

Comment:
The `err == nil` can be simplified further if the preceding error check is broken down!
I believe that it would make these checks more readable and easy to understand.

Reply:
sure, will make the change